lib/ssh_scan/config_loader.rb

```diff
@@ -0,0 +1,31 @@
+require 'logger'
+
+require 'ssh_scan/config_loader/yaml'
+
+module SSHScan
+  module ConfigLoader
+    def self.load(config, class_name = nil, mode = 'file', logger = nil)
+      case mode
+        when 'file'
+          ext = File.extname(config)
+          case ext
+            when ".yaml", ".yml"
+              loaded_config = SSHScan::ConfigLoader::Yaml.new(config, class_name, 'file', logger)
+              return loaded_config.checkedOpts
+            else
+              logger.warn("no support for this kind of file yet: #{ext}") if logger
+          end
+        when 'string'
```

Comment:
Also, I think this looks for a string of "string".  If you want to match on a class type, you need to say String unquoted.